lib/fizzy/instantiate.cpp

```diff
@@ -203,7 +203,7 @@ std::tuple<bytes_ptr, Limits> allocate_memory(const std::vector<Memory>& module_
         }
 
         // NOTE: fill it with zeroes
-        bytes_ptr memory{new bytes(memory_min * PageSize, 0), bytes_delete};
+        bytes_ptr memory{new bytes(uint32_t{memory_min * PageSize}, 0), bytes_delete};
```

Comment:
@gumb0 can you double check?

Reply:
Note that this is non-functional change. The `uint32_t{}` explicitly states that the multiplication is done with 32-bit precision. This will not compile if any of multiplication operands has bigger type.

Reply:
Is this change still needed with disabled `bugprone-implicit-widening-of-multiplication-result`?

Reply:
It is not. I started "fixing" these but it turned out there are too many cases for constant multiplication. I reported the bug about the check but it is not worth it in current form.

Reply:
Well turns out we have a bug on master here, because trying to allocate 4Gb actually overflows and allocates 0.

https://github.com/wasmx/fizzy/pull/805

Reply:
The `uint32_t{}` is my proposed why of silencing the warning. I can revert these 2 changes from this PR, especially if there is a fix now in a side PR.

Reply:
I would revert them if warning is disabled anyway.